src/AutorestSwift/View Models/KeyValueViewModel.swift

```diff
@@ -32,15 +32,32 @@ import Foundation
 struct KeyValueViewModel {
     let key: String
     let value: String
+    let valueNilable: Bool
 
-    init(from param: Parameter, with model: CodeModel) {
-        self.key = param.serializedName!
-        if let constant = param.schema as? ConstantSchema {
-            self.value = "\"\(constant.value.value)\""
+    init(from param: Parameter, with model: CodeModel, and operation: Operation, using key: String? = nil) {
+        self.key = key ?? param.serializedName!
+
+        if let constantSchema = param.schema as? ConstantSchema {
+            let isString: Bool = constantSchema.valueType.type == AllSchemaTypes.string
+            let val: String = constantSchema.value.value
+
+            self.value = isString ? "\"\(val)\"" : "\(val)"
+            self.valueNilable = false
```

Comment:
```suggestion
            let val = constantSchema.value.value
            if constantSchema.valueType.type == AllSchemaTypes.string {
              self.value = "\"\(val)\""
            } else {
              self.value = "\(val)" 
            }
            self.valueNilable = false
```

Fewer boolean variables should make it a little clearer.